### PR TITLE
Introduce a Helm chart for Grafana Agent

### PR DESCRIPTION
**Pull request by rfratto**
This PR introduces a Helm chart for Grafana Agent which primarily focuses on Grafana Agent Flow. It is upstreamed from rfratto/agent-helm-chart.

The following will be done in follow up PRs: 

- [ ] Documentation 
- [ ] Update the build image to include additionally needed tooling for regenerating the README and tests 
- [ ] Signing the Helm charts 
- [ ] Linking to this chart from grafana/helm-docs

If you're reviewing this PR, I suggest only looking at the first commit. The other commits bring in generated files and the bulk of the size of this PR.

### CODE REVIEW

**Comment by rfratto** [comment]
(Shoutout to @Logiraptor for having a chat with me about Mimir's Helm charts, which I've hopefully been able to incorporate some learnings into here)

**Comment by osg-grafana** [comment]
Related: https://github.com/grafana/website/pull/11243

**Comment by osg-grafana** [comment]
> (Shoutout to @Logiraptor for having a chat with me about Mimir's Helm charts, which I've hopefully been able to incorporate some learnings into here)

:) Please start by following this dir structure: https://github.com/grafana/mimir/tree/main/docs/sources

**Comment by rfratto** [comment]
I'm not sure if we want to cover documentation in this PR. I feel like documenting the Helm chart will probably be its own, separate undertaking

**Review by captncraig — APPROVED** [review]
Overall I love it. I'd like to see at least a readme in grafana/helm-charts that points people here.

File: operations/helm/charts/grafana-agent/README.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+| configReloader.customArgs | list | `[]` | Override the args passed to the container. |
+| configReloader.enabled | bool | `true` | Enables automatically reloading when the agent config changes. |
+| configReloader.image.repository | string | `"weaveworks/watch"` | Repository to get config reloader image from. |
+| configReloader.image.tag | string | `"master-5fc29a9"` | Tag of image to use for config reloading. |
+| controller.podAnnotations | object | `{}` | Extra pod annotations to add. |
+| controller.replicas | int | `1` | Number of pods to deploy. Ignored when controller.type is 'daemonset'. |
+| controller.tolerations | list | `[]` | Tolerations to apply to Grafana Agent pods. |
+| controller.type | string | `"daemonset"` | Type of controller to use for deploying Grafana Agent in the cluster. Must be one of 'daemonset', 'deployment', or 'statefulset'. |
+| controller.updateStrategy | object | `{}` | Update strategy for updating deployed Pods. |
+| controller.volumeClaimTemplates | list | `[]` | volumeClaimTemplates to add when controller.type is 'statefulset'. |
```
[the remark this answers is not in the document]
Comment (rfratto) [reply operations/helm/charts/grafana-agent/README.md:63]: > extra secrets? env vars? configMaps? Not sure if all are needed, but some probably will be.

All of these are supported already:

* Extra secrets and configMaps can be provided via `controller.volumes.extra` and `agent.mounts.extra`
Extra env vars can be provided via `agent.extraEnv`

File: operations/helm/charts/grafana-agent/README.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+| controller.volumes.extra | list | `[]` | Extra volumes to add to the Grafana Agent pod. |
+| fullnameOverride | string | `nil` | Overrides the chart's computed fullname. Used to change the full prefix of resource names. |
+| image.pullPolicy | string | `"IfNotPresent"` | Grafana Agent image pull policy. |
+| image.pullSecrets | list | `[]` | Optional set of image pull secrets. |
+| image.repository | string | `"grafana/agent"` | Grafana Agent image repository. |
+| image.tag | string | `nil` | Grafana Agent image tag. When empty, the Chart's appVersion is used. |
+| nameOverride | string | `nil` | Overrides the chart's name. Used to change the infix in the resource names. |
+| rbac.create | bool | `true` | Whether to create RBAC resources for the agent. |
+| serviceAccount.annotations | object | `{}` | Annotations to add to the created service account. |
```
[the remark this answers is not in the document]
Comment (rfratto) [reply operations/helm/charts/grafana-agent/README.md:72]: `serviceAccount.annotations` only applies to the service account. Do you think it should apply to all RBAC?

**Comment by rfratto** [comment]
I'm going to merge this as-is once the CI passes. I'd like to leave docs to a follow up PR since this PR is big enough already :)

**Comment by osg-grafana** [comment]
> I'm not sure if we want to cover documentation in this PR. I feel like documenting the Helm chart will probably be its own, separate undertaking

Yes, that is what we have learned as well.

**Pull request merged**